### PR TITLE
[clipboard] add native implementation for expo-clipboard

### PR DESCRIPTION
**Pull request by cruzach**
# Why

closes https://linear.app/expo/issue/ENG-1030/extract-native-code-to-expo-clipboard

# How

added getStringAsync, setString, and listener methods

# Test Plan

Tested in NCL in Expo Go on android and ios

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-clipboard/package.json [hunk 1]
```diff
@@ -29,6 +29,9 @@
   "author": "650 Industries, Inc.",
   "license": "MIT",
   "homepage": "https://docs.expo.io/versions/latest/sdk/clipboard",
+  "dependencies": {
+    "@unimodules/core": "~7.1.0"
```
Comment (brentvatne) [inline packages/expo-clipboard/package.json:33]: remove

**Review by brentvatne — APPROVED** [review]

**Review by EvanBacon — COMMENTED** [review]
It seems like a lot of the new things we've been doing with expo-dev-menu aren't represented here. Some initial things I'm seeing:
- native iOS name is EXClipboard.podspec, whereas `expo-dev-menu` uses `expo-dev-menu.podspec` and the ios folder directly
- `s.source_files = 'ios/Plugin/**/*.{swift,h,m,c,cc,mm,cpp}'` account for all files upfront
- not using swift
- `s.swift_version`

**Comment by brentvatne** [comment]
@EvanBacon - i think expo-dev-menu is a bit of a one off right now, we haven't established those conventions for all new modules yet. expo-dev-menu isn't a unimodule / "expo module", for example. i think we want to move to using swift for our expo modules but we need to have established patterns for doing this first. @tsapeta has explored it a bit and has some nice ideas.

**Comment by tsapeta** [comment]
@EvanBacon, as Brent said, I'm exploring another way to write Swifty Expo modules, so I think for now it's fine to use ObjC for the new code on the iOS side. In terms of native iOS name convention, I'm in favor of `ExpoClipboard` — this is clear and fits better to the other iOS frameworks. I already let myself to go that way in `expo-modules-core` so it's called `ExpoModulesCore` 😅

**Review by tsapeta — APPROVED** [review]

**Review by brentvatne — APPROVED** [review]

**Pull request merged**